packages/react-reconciler/src/ReactFiberRoot.js

```diff
@@ -37,6 +38,10 @@ export type FiberRoot = {
   pendingContext: Object | null,
   // Determines if we should attempt to hydrate on the initial mount
   +hydrate: boolean,
+  // TODO: This belongs in the renderer, which suggests we should use something
+  // other than the UpdateQueue type, since it would be duplicated across
+  // packages regardless.
+  deferredCommits: UpdateQueue<null> | null,
```

Comment:
Can you explain conceptually what this is and why we need it? Why isn't this just part of the normal update queue? What is it about this that requires a separate data structure?

I was expecting these to go on the update queue of the root and just not committed.

Reply:
Hmm, that's how I did it in https://github.com/facebook/react/pull/11232... The root state contains a `deferred` field that's true if the tree shouldn't be committed yet. Don't know why I didn't do that here again. I think maybe I thought it would conflict with the new deterministic updates behavior, but that doesn't sound right. I'll update.

Reply:
Ah, I think it was this: the current `renderRoot` API returns `finishedWork` if the tree completes, but there's no way to say "this tree has no more work left but also it isn't ready to commit yet."

I wish we could return multiple values from `renderRoot`. Like a tuple of `(finishedWork, deferCommit, didError, errorObject)`

Reply:
I don't see how this follows. There's something off in the model here. Both the deferred flag and this separate list models something much more powerful (arbitrary updates being deferred and placed in separate order) than what can actually happen. Makes it hard to think about and track bugs

Reply:
The tricky part is that a root can be deferred at multiple levels. When the first one is flushed, the second level is still blocked from committing.

After talking offline, I'll update to use an array of expiration times instead of using the UpdateQueue type.